netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -179,6 +179,9 @@ typedef struct _net_ebpf_bpf_sock_addr
     TOKEN_ACCESS_INFORMATION* access_information;
     uint64_t process_id;
     uint32_t flags;
+    net_ebpf_extension_hook_id_t hook_id;
```

Comment:
For my understanding, can you explain the need to add hook_id?

Reply:
Setting the redirect context is only supported when the CONNECT_REDIRECT layer callout is invoked (which subsequently invokes an ebpf program). I needed to add this to track which layer is getting invoked, so that the helper function could return error if this is called at the incorrect layer. I did not see anything existing which could do this, so I added this field. (Please let me know if you see another way to check this without adding this field).

Reply:
Thanks @matthewige for explaining.